AppSandboxFileAccess/Classes/AppSandboxFileAccess.m

```diff
@@ -160,15 +205,11 @@ - (BOOL)accessFileURL:(NSURL *)fileUrl withBlock:(AppSandboxFileAccessBlock)bloc
 	
 	// if we have no bookmark data, we need to create it, this may be because our bookmark data was stale, or this is the first time being given permission
 	if (persist && !bookmarkData) {
-		[self persistPermissionURL:allowedUrl];
+		bookmarkData = [self persistPermissionURL:allowedURL];
 	}
 	
-	// execute the block with the file access permissions
-	@try {
-		[allowedUrl startAccessingSecurityScopedResource];
-		block();
-	} @finally {
-		[allowedUrl stopAccessingSecurityScopedResource];
+	if (block && allowedURL && bookmarkData) {
+		block(allowedURL, bookmarkData);
```

Comment:
Actually that's pretty pointless to even raise an exception at this point, because it's something that shouldn't happen and is only happening for a stale bookmark because I'm incorrectly renewing the bookmark. So we should assume `bookmarkData` and `allowedURL` will not be nil at this point and reduce the if to `if (block)`.